crates/interpreter/src/exec.rs

```diff
@@ -1021,12 +1028,13 @@ impl<'m> Thread<'m> {
     }
 
     fn pop_values(&mut self, n: usize) -> Vec<Val> {
-        let mut values = Vec::new();
-        for _ in 0 .. n {
-            values.push(self.pop_value());
-        }
-        values.reverse();
-        values
+        self.label().values_cnt -= n;
+        self.only_pop_values(n)
+    }
+
+    fn only_pop_values(&mut self, n: usize) -> Vec<Val> {
```

Comment:
I expect this function to disappear once `values_cnt` disappears, since it becomes equivalent to `pop_values`.